src/net.cpp

```diff
@@ -2525,9 +2526,20 @@ void CConnman::AddNewAddresses(const std::vector<CAddress>& vAddr, const CAddres
     addrman.Add(vAddr, addrFrom, nTimePenalty);
 }
 
-std::vector<CAddress> CConnman::GetAddresses()
+std::vector<CAddress> CConnman::GetAddresses(Optional<Network> requestor_network)
 {
-    return addrman.GetAddr();
+    if (requestor_network) {
```

Comment:
This might be an insane edge case, but if `pfrom.addr.GetNetwork()` is `NET_UNROUTABLE`, the value will be `0` (and falsey ?) and break the intent of this logic. I'm not sure how an attacker could use an unroutable address to break your cache though so probably moot.

Reply:
Are you sure Optional(0) would be ``false``? Would be useful to double-check the optional spec, but I expected that any value present (including 0) would give ``true``

Reply:
Confirmed, you are right. Thanks!